### PR TITLE
feat(misconf): variable support for Terraform Plan

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description

This PR adds support for input variables that are stored in Terraform Plan.

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/7222

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: magefiles/magefile.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ var (
 	}
```
Comment (simar7) [inline magefiles/magefile.go:30]: was this file changed because of a linter issue?

File: pkg/iac/scanners/terraformplan/snapshot/plan.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	plan := Plan{
+		variableValues: make(map[string]DynamicValue),
+	}
+
+	for k, v := range rawPlan.Variables {
+		if len(v.Msgpack) == 0 { // len(0) because that's the default value for a "bytes" in protobuf
+			return nil, fmt.Errorf("dynamic value does not have msgpack serialization")
+		}
```
Comment (simar7) [inline pkg/iac/scanners/terraformplan/snapshot/plan.go:58]: I'm not sure if I understand the comment, we check the len to be zero to ensure that what we are unpacking is msgpack right?

File: magefiles/magefile.go [hunk 3]
```diff
@@ -173,6 +177,13 @@ func Protoc() error {
 				return err
 			}
 		}
+
+		for _, file := range protoFiles {
+			if err := sh.RunV("protoc", ".", "paths=source_relative", "--go_out", ".", "--go_opt",
+				"paths=source_relative", file); err != nil {
+				return err
+			}
+		}
```
Comment (simar7) [inline magefiles/magefile.go:186]: If it is run in the container, would we need to update https://github.com/aquasecurity/trivy/blob/5e20c115fee9196943a3715c4f85bb3eac9d7d66/Dockerfile.protoc as well?

Comment (simar7) [reply magefiles/magefile.go:186]: See the alternate logic here https://github.com/aquasecurity/trivy/pull/7228/files#diff-d7a1577b2d546e34aaf816bdd1b83edef28a66a555f049767bafa83268b7efbbR190-R195

File: pkg/iac/scanners/terraformplan/snapshot/plan.go (hunk 2 above)
Comment (nikpivkin) [reply pkg/iac/scanners/terraformplan/snapshot/plan.go:58]: Terraforom stores cty values in msgpack encoding . If the byte length is 0, it means a invalid value is represented. I looked up the [implementation](https://github.com/hashicorp/terraform/blob/main/internal/plans/planfile/tfplan.go#L449) in Terraform.

File: magefiles/magefile.go (hunk 3 above)
Comment (nikpivkin) [reply magefiles/magefile.go:186]: `mage:protoc` command builds an image and runs it. The dockerfile does not need to be updated.

**Review by simar7 — APPROVED** [review]

**Pull request merged**